source/extensions/filters/common/rbac/engine_impl.cc

```diff
@@ -8,31 +8,52 @@ namespace RBAC {
 
 RoleBasedAccessControlEngineImpl::RoleBasedAccessControlEngineImpl(
     const envoy::config::filter::http::rbac::v2::RBAC& config, bool disabled)
-    : engine_disabled_(disabled),
-      allowed_if_matched_(disabled ||
-                          config.rules().action() ==
-                              envoy::config::rbac::v2alpha::RBAC_Action::RBAC_Action_ALLOW) {
+    : engine_disabled_(disabled), config_(config) {
   if (disabled) {
     return;
   }
 
   for (const auto& policy : config.rules().policies()) {
     policies_.emplace_back(policy.second);
   }
+
+  for (const auto& policy : config.permissive_rules().policies()) {
+    permissive_policies_.emplace_back(policy.second);
+  }
 }
 
 RoleBasedAccessControlEngineImpl::RoleBasedAccessControlEngineImpl(
     const envoy::config::filter::http::rbac::v2::RBACPerRoute& per_route_config)
     : RoleBasedAccessControlEngineImpl(per_route_config.rbac(), per_route_config.disabled()) {}
 
 bool RoleBasedAccessControlEngineImpl::allowed(const Network::Connection& connection,
-                                               const Envoy::Http::HeaderMap& headers) const {
+                                               const Envoy::Http::HeaderMap& headers,
+                                               EnforcementMode mode) const {
   if (engine_disabled_) {
```

Comment:
I'm still don't understand the purpose of this boolean. Can we just use the logic with the next two `if` you've just added?

Reply:
I also feel it's not needed, it could be inferred from whether rules is set; removed it in https://github.com/envoyproxy/envoy/pull/3554/commits/c4f5e47da63c8df9f35fe023b6e326ce1ef2c0b7.  @rodaine let's know if I miss anything here.